### PR TITLE
Exposed the recursion limit value as a config option

### LINKED ISSUE #12253: expose the recursion limit value as a config option
### Related problem

I have a script that via recursion, iterates trough the pages of a youtube playlist via the youtube api. One youtube playlist has around 60 pages, and therefore I get the `Recursion limit (50) reached` error, since the limit is set to 50.

### Describe the solution you'd like

a configuration option to set this value to our needs

### Describe alternatives you've considered

_No response_

### Additional context and details

_No response_

### PR DESCRIPTION
**Pull request by merelymyself**
# Description

Closes #12253.

Exposes the option as "recursion_limit" under config.

# User-Facing Changes

The config file now has a new option!

# After Submitting

Nothing else...? Do let me know if there's something I've missed!

### CODE REVIEW

File: crates/nu-protocol/src/config/mod.rs [hunk 1]
```diff
@@ -61,6 +61,7 @@ pub struct Config {
     pub footer_mode: FooterMode,
     pub float_precision: i64,
     pub max_external_completion_results: i64,
+    pub recursion_limit: i64,
```
Comment (sholderbach) [inline crates/nu-protocol/src/config/mod.rs:64]: We may want to guard against underflow with negative values.

**Review by sholderbach — COMMENTED** [review]
Great to see you around!

Thanks for giving this a go.

We may in the future want to get rid of this option again to maintain a reasonable complexity of `Config`

**Comment by fdncred** [comment]
wow, nice to see you back again @merelymyself. It's been a while.

**Comment by fdncred** [comment]
Let's give this a whirl to see if it helps these edge case scenarios.

**Pull request merged**